yasnippet.el

```diff
@@ -3286,58 +3300,29 @@ If so cleans up the whole snippet up."
 
 ;; Apropos markers-to-points:
 ;;
-;; This was found useful for performance reasons, so that an
-;; excessive number of live markers aren't kept around in the
-;; `buffer-undo-list'. However, in `markers-to-points', the
+;; This was found useful for performance reasons, so that an excessive
+;; number of live markers aren't kept around in the
+;; `buffer-undo-list'.  However, in `markers-to-points', the
 ;; set-to-nil markers can't simply be discarded and replaced with
-;; fresh ones in `points-to-markers'. The original marker that was
-;; just set to nil has to be reused.
+;; fresh ones in `points-to-markers'.  The original marker that was
+;; just set to nil has to be reused (why?).
```

Comment:
> Sorry, I'm not very familiar with it, where is the equivalent code?

Uhhh, nowhere. That was kind of the point i guess. I messed up, I haven't implemented undo (specifically snippet-revival( yet, but I remember seeing it as merely reactivating the active field overlay, since all the markers and marker positions were still valid. 

So, to summarize, try this. Remove the code deleting and reinstating the markers on snippet clean/revival respectively, run the automated tests and try out `garbage-collect`. You should be OK.